samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -267,20 +273,41 @@ public TestRunner addOutputStream(CollectionStream stream) {
     return this;
   }
 
+
   /**
    * Utility to run a test configured using TestRunner
+   *
+   * @param timeout in ms for the Test to Run
+   * @throws InterruptedException if the current thread was interrupted
+   * @throws ExecutionException if the samza job threw an exception
+   * @throws TimeoutException if the wait timed out while running Samza job
```

Comment:
I think rather than throwing all the `Exceptions` back to user, would be better to throw one custom-defined samza exception(`SamzaException`, `TestRunnerException` etc)

Reply:
using Samza's waitForFinish(timeout) should remove this, although will keep in mind